final/background/spoofax.tex

```diff
@@ -1,57 +1,89 @@
-execution. Dynamic semantics define what the behaviour is of a
-program upon execution. In Spoofax, the dynamic semantics can be
-defined with either Stratego or a DSL called DynSem.
+\item \textbf{\hyperref[ssec:syntax-def]{Syntax Definition}:} The first aspect
+  concerns what textual representations of a program are syntactically
+  valid. Its specification consists of a grammar specified with context-free
+  grammar productions. A parser provides an implementation of this
+  specification, by mapping a textual representation of a program to an abstract
+  syntax tree (AST) representation. In Spoofax, the syntax is declared with a
+  domain specific language (DSL) called SDF.
+\item \textbf{Static Semantics:} A parsed AST then optionally goes through
+  static analysis (type checking, name binding and variable scoping), to test if
+  the program is well-formed. Static analysis is the implementation of an aspect
+  called \textit{static semantics}. The static semantics of a language can be
+  formally specified by \textit{type and/or name binding rules}. Spoofax
+  provides two DSLs that can specify the two distinct parts of the
+  specification: the TS Type Specification language and the NaBL name binding
+  language.
+
+  This language aspect will not be discussed in this section. However, the
+  original research report contains a section that does discuss this aspect, and
+  can be found in \cref{ssec:a-static-analysis}.
+\item \textbf{Term Rewriting or Program Transformation:} Optionally, a
+  well-formed AST can then be transformed, for example for desugaring or
+  optimization. This execution step falls under the aspect of \textit{term
+    rewriting} and can be formally specified by \textit{term rewrite rules}. For
+  the specification of this aspect, Spoofax provides a DSL called Stratego.
+
+  This aspect, too, will not be discussed in this section. The relevant section
+  in the research report can be found in \cref{ssec:a-term-rewrite}.
+\item \textbf{\hyperref[ssec:dynamic-semantics]{Dynamic Semantics}:} Next the
+  optionally transformed AST is either compiled or interpreted, thereby
+  providing a means of execution. The aspect of \textit{dynamic semantics}
+  defines what the behaviour is of a program upon execution. The formal
+  specification of this aspect can be done with \textit{reduction rules},
+  although there are other ways to accomplish this: for example, interpretation
+  can be seen as the transformation of a program to a single value. In Spoofax,
+  the dynamic semantics can be defined with either Stratego or a DSL called
+  DynSem.
 \end{enumerate}
 
-This section concludes with a discussion on the other
-aspect of a language: its integrated development environment
-(IDE). Spoofax provides IDE support by means of its Editor Services.
+\begin{figure}[t]
+  \centering
+  \includegraphics[width=\textwidth]{spoofax-relations}
+  \caption{The relations between the aspects, specification and implementation
```

Comment:
Indeed when you look at it like that it's fine, but when you understand it as the specification and implementation of the aspectS then it should be plural.